### PR TITLE
[Tooling] Allow `upload_to_play_store` to retry on Google API failure

### PR DESCRIPTION
**Pull request by AliSoftware**
This is a PR to workaround a Google API error we encounter randomly when trying to upload new builds to the PlayStore in parallel

⚠️  Note: I've made this PR target `release/19.2` because it's most likely that the next time we will use this lane will be during release duties of `19.2`

### Context

When we build a new alpha/beta or the final release and try to upload it to the PlayStore, sometimes the build ends failing on the very last step with this error message ([example](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Android/28983/workflows/9dfcdff2-5b0a-44bd-ad7a-b3257c65f0f5/jobs/118953)):

```
[…]
| 17   | Switch to upload_build_to_play_store lane                     | 0           |
| 💥   | upload_to_play_store                                          | 31          |
+------+---------------------------------------------------------------+-------------+

[09:49:27]: fastlane finished with errors

[!] Google Api Error: Invalid request - This Edit has been deleted.

Exited with code exit status 1
```

Retrying the build ("Rerun from failed"on CircleCI) typically fixes it every time.

### Why

We are not entirely sure why this happens, but it seems a bit random and likely due to something of a race condition happening between CI and the Google API.

There's an [old GitHub Issue on fastlane about it](https://github.com/fastlane/fastlane/issues/13689), but it has been auto-closed and likely won't be fixed anytime soon. [One of the comment in this issue](https://github.com/fastlane/fastlane/issues/13689#issuecomment-439217502) suggests that the issue might happen when we try to run the lane in parallel. Given that we indeed parallelized the build-and-upload of WP alpha, WP beta and JP beta, it seems quite likely that there could be a race condition there with each trying to do an "Edit" at roughly the same time.

An "Edit" on PlayStore API is like a transaction in a DB: the API requires you top [begin an Edit](https://github.com/fastlane/fastlane/blob/master/supply/lib/supply/uploader.rb#L7), then make your changes, then [commit the Edit](https://github.com/fastlane/fastlane/blob/master/supply/lib/supply/uploader.rb#L43) to validate them; so indeed it seems likely that the job for WP alpha opens an Edit, and the one for WP beta also opens one, inherently cancelling the WP-alpha one, leading to the error.

On the other hand, I've seen cases where the **very first** of the 3 jobs is the one that fails (see the order of the messages in the Slack notifications ref. p1644832168536729-slack-C5ALGJYEP ), which makes me wonder if that situation above is really what's happening

### Workaround

In any case, since retrying the job makes it pass, I've decided to implement an auto-retry mechanism on the upload step itself, to avoid us having to be surprised by the error notification, have to re-run the job manually (only when the other 2 have finished because the UI to "rerun from failed" needs so), and also only retry the upload (instead of having to restart the whole build again).

This might not be perfect (it's only a workaround, after all), but seems an easy enough to implement (as opposed to having to implement some semaphore mechanism to sync the build or completely split and rearrange the job dependencies just for this) and hopefully enough to reduce the amount of those API errors in the future.

### CODE REVIEW

File: fastlane/Fastfile
Comment (AliSoftware) [inline fastlane/Fastfile:597]: Given this API Error likely happens because multiple jobs (WP alpha / WP beta / JP beta) are trying to { open an Edit, make changes to the track, commit the Edit} in parallel, it seems relevant to wait a bit before retrying, to avoid having each parallel job otherwise constantly cancelling each other.

The 2mn delay is somewhat arbitrary, as I figured waiting for only 5s would be too few (risking to cancel the other parallel job — the one which has cancelled this Edit — to fail in turn, and having each job cancelling each other) and 5mn a bit too long. Let's see how this goes.

File: fastlane/Fastfile
Comment (AliSoftware) [inline fastlane/Fastfile:598]: TIL the `retry` keyword in ruby btw.

FYI, it's only valid inside a `rescue` block and will retry/re-run the whole `begin…end` block

File: fastlane/Fastfile
Comment (AliSoftware) [inline fastlane/Fastfile:600]: If the condition above didn't evaluate to true (the raised exception was not about Google Api Error, or it was but the `retry_count` was 0), then just re-raise the exception (just like if we didn't rescue it)

**Comment by AliSoftware** [comment]
Alternative workaround that could be worth keeping in mind when we'll migrate this to Buildkite's `pipeline.yml` (Cc @jkmassel): move the `upload_to_playstore` last step of each 3 jobs _after_ the joining of the 3 jobs back into GitHub release, to keep the 3 builds in parallel but make the 3 aab uploads **sequential** (and only after the 3 builds have finished similar to how we wait for them to upload the aabs to GitHub Release)

That will require a bit more rearranging of the lanes and CI pipelines and step dependencies so makes more sense to wait for Buildkite migration to tackle that more involved solution, and the workaround should hopefully do in the meantime.

**Review by ParaskP7 — APPROVED** [review]
👋 @AliSoftware !

I have reviewed this PR, it LGTM, good job! 🌟

We can indeed try this solution for this, and any subsequent releases, and then see how it will work or whether, for some unknown reason, it won't. I think the `2' mins` waiting time until retrying it again for 2 times is a good start and seems balanced enough. It's a 🟢 light from me.

PS: I do like the `Buildkite` solution you suggested afterwards. This seems a much better way to deal with this problem. However, as you said, let's wait for the migration to finish before working on improving the pipeline.

**Pull request merged**
